From 713144e6193e0e167842355a231cb807fae29cd3 Mon Sep 17 00:00:00 2001
From: RajaniRanjan <rajani.ranjan@intel.com>
Date: Tue, 1 Feb 2022 09:37:26 +0530
Subject: [PATCH] keyboard input support to move cursor by left and right keys

PGP keyboard input keys for left, right, up, down.

Tracked-On: OAM-100842
Signed-off-by: RajaniRanjan <rajani.ranjan@intel.com>
---
 client/include/vInputClient.h |  9 +++++++++
 client/src/vInputClient.c     | 21 +++++++++++++++++++++
 2 files changed, 30 insertions(+)

diff --git a/client/include/vInputClient.h b/client/include/vInputClient.h
index e95772e2..9b9b71c0 100644
--- a/client/include/vInputClient.h
+++ b/client/include/vInputClient.h
@@ -1,6 +1,15 @@
 #define XRES_MAX 600.00
 #define YRES_MAX 960.00
 #define LTRB_Y   292.00
+#define LG_LEFT_ARROW 57419
+#define LG_RIGHT_ARROW 57421
+#define LG_UP_ARROW 57416
+#define LG_DOWN_ARROW 57424
+#define LINUX_LEFT 105
+#define LINUX_RIGHT 106
+#define LINUX_UP 103
+#define LINUX_DOWN 108
+
 typedef enum {PRESS, RELEASE, MOVE, KEYDOWN, KEYUP} action;
 //typedef enum {ROTATION_0, ROTATION_1} rotation;
 
diff --git a/client/src/vInputClient.c b/client/src/vInputClient.c
index 0aa50f7a..07377315 100644
--- a/client/src/vInputClient.c
+++ b/client/src/vInputClient.c
@@ -72,6 +72,27 @@ bool vinput_touch(struct vinputEvent ev){
 	if (mqId == -1) {
 		return false;
 	}
+	switch(ev.keys){
+                case LG_LEFT_ARROW:
+                        ev.keys = LINUX_LEFT;
+                        break;
+
+                case LG_RIGHT_ARROW:
+                        ev.keys = LINUX_RIGHT;
+                        break;
+
+                case LG_UP_ARROW:
+                        ev.keys = LINUX_UP;
+                        break;
+
+                case LG_DOWN_ARROW:
+                        ev.keys = LINUX_DOWN;
+                        break;
+
+                default:
+                        break;
+                }
+
 	if (ev.fullscreen == 1)
 	{
 		switch (ev.rotation) {
-- 
2.25.1

